sdk/go/internal/querybuilder/querybuilder.go

```diff
@@ -82,26 +82,38 @@ func (s *Selection) build(ctx context.Context) (string, error) {
 	if err := s.marshalArguments(ctx); err != nil {
 		return "", err
 	}
-	fields := []string{
-		"query",
-	}
+
+	var b strings.Builder
+	b.WriteString("query")
+
 	for _, sel := range s.path() {
-		q := sel.name
+		b.WriteRune('{')
+
+		if sel.alias != "" {
+			b.WriteString(sel.alias)
+			b.WriteRune(':')
+		}
+
+		b.WriteString(sel.name)
+
 		if len(sel.args) > 0 {
-			args := make([]string, 0, len(sel.args))
+			b.WriteRune('(')
+			i := 0
 			for name, arg := range sel.args {
-				args = append(args, fmt.Sprintf("%s:%s", name, arg.marshalled))
+				if i > 0 {
+					b.WriteString(", ")
+				}
+				b.WriteString(name)
+				b.WriteRune(':')
+				b.WriteString(arg.marshalled)
+				i++
 			}
-			q += "(" + strings.Join(args, ", ") + ")"
-		}
-		if sel.alias != "" {
-			q = sel.alias + ":" + q
+			b.WriteRune(')')
 		}
-		fields = append(fields, q)
 	}
 
-	q := strings.Join(fields, "{") + strings.Repeat("}", len(fields)-1)
-	return q, nil
+	b.WriteString(strings.Repeat("}", len(s.path())))
```

Comment:
nit: Could pull `s.path()` out and reuse it between here and the loop so we only allocate it once.

Reply:
Sounds good @vito

Reply:
Done!